File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/composite/FormTwoButtonsTests.java
package com.epam.jdi.uitests.testing.unittests.tests.composite;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.pageobjects.pages.IContactForm;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.Contact.DEFAULT;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.contactFormPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.checkResult;

/**
 * Created by Dmitry_Lebedev1 on 10/15/2015.
 */
public class FormTwoButtonsTests extends InitTests {
    Supplier<IContactForm> form = () -> contactFormPage.contactFormTwoButtons;

    @BeforeMethod
    public void before(final Method method) {
        isInState(CONTACT_PAGE, method);
    }

    @Factory
    public Object[] textTests() {
        return new Object[]{
                new FormTests(form)
        };
    }

    @Test
    public void submitSpecButtonStringTest() {
        form.get().submit(DEFAULT, "calculate");
        checkResult("Summary: 3");
    }
}

File: Python/Test/jdi_uitests_webtests/test/composite/form_two_buttons_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.entities.contact import Contact
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class FormTwoButtonsTests(InitTests):
    form = EpamJDISite.contact_form_page.contact_form_two_buttons
    contact = Contact("Ivan", "Ivanov", "Smart Man")

    def setUp(self):
        Preconditions.CONTACT_PAGE.is_in_state()

    def test_submit_spec_button_string(self):
        self.form.submit_form(self.contact, "calculate")
        Assert.assert_equal(EpamJDISite.contact_form_page.result.get_text(), "Summary: 3")